### DISCUSSION TITLE
How do I get the name of a VoucherFileConnection?

### DISCUSSION THREAD

**Discussion by SWICRIOT** [FortnoxAB/csharp-api-sdk#296] [Q&A] [answered]
A SupplicerInvoiceFileConnection has a FileId and a Name. A VoucherFileConnection does not have a Name-property. When downloading the file directly in Fortnox web app it obviously has a proper file name.

I don't see any method in the ArchiveConnector either to get the information of a file.

**Comment by adamelfstrom** [comment]
There is no `name` field on the `VoucherFileConnection`, but there is a `VoucherDescription`. Does that help you?

Ref: https://apps.fortnox.se/apidocs#tag/VoucherFileConnections

**Reply by SWICRIOT** [reply]
Unfortunately not, it does not contain relevant info about the attachment. I look at the official API documentation as well and I don't see anything there. Must be mistakenly forgotten? If I don't have the name I don't even know what type of file it is unless I use a third-party library to analyse the file contents.

**Comment by SWICRIOT** [comment]
I got an answer from Fortnox now that the filename is contained within the _content-disposition_-header when using the Inbox-API. It's probably the same for the Archive-API. But then when using this SDK, I only get a byte array for the file content.

Regardless, it seems like a mistake that SupplierInvoiceFileConnection has a name and VoucherFileConnection does not. Even if I theoretically could get the name by downloading the file through the SDK, that would mean that I would have to download all the files just to get a list of their names.

**Comment by SWICRIOT** [comment, marked as answer]
Fortnox has now added the Name property. See PR https://github.com/FortnoxAB/csharp-api-sdk/pull/297.
